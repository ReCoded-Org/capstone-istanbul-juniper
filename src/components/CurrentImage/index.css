.puzzlePage__puzzleBox__CurrentImage{
  margin: 0;
  padding: 0;
  height: 100%;
  width: 100%;
  overflow: hidden;
  user-select: none;
  font-family: 'Raleway', sans-serif;
  display: flex;
  flex-direction: column;
  background: #f0f0f0;
  cursor: url('https://uploads.codesandbox.io/uploads/user/b3e56831-8b98-4fee-b941-0e27f39883ab/Ad1_-cursor.png') 39 39,
    auto;

}

.selectedTile {
  background-color: green;
}