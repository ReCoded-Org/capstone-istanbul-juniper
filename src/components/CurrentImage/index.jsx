import React,{ useState, useEffect } from "react";
import {Col, Row} from "antd"
import "./index.css"

const CurrentImage = () => {
    const [generatedTiles, setGeneratedTiles] = useState([])
    const [tileSelected, setTileSelected] = useState({id: null, selected: false})
    const [isSolved, setIsSolved] = useState(false) // if unsolved it displays the tiles and when solved it displays the full image
    const generateTiles = () => {
        let newTilesArray = [];
        for (let i=0; i<6; i++){
            let newTile = {id:i, selected: false} // These tiles are puzzle pieces
            newTilesArray.push(newTile)
        }
        let length = newTilesArray.length
        for(let i=0;i<length;i++){
            let random = Math.floor(Math.random()*(length-1))
            let randomItemArray = newTilesArray.splice(random,1)
            newTilesArray.push(randomItemArray[0])
        }
        setGeneratedTiles(newTilesArray)
    }

    useEffect(() => {
        generateTiles();
    }, [])

    useEffect(() => {
        checkSolution()
    }, [generatedTiles])

    const handleSelect = (currId) => {
        if (tileSelected.selected) {
            const prevId = tileSelected.id
            swapTiles(prevId, currId)
            
        } else {
            setTileSelected({id: currId, selected: true })
            const tileIndex = generatedTiles.findIndex(tile => tile.id === currId)
            generatedTiles[tileIndex] = {...generatedTiles[tileIndex], selected: true}
        }
    }

    const swapTiles = (id1, id2) => {
        let newTiles = [...generatedTiles]
        let index1 = generatedTiles.findIndex(tile => tile.id === id1)
        let index2 = generatedTiles.findIndex(tile => tile.id === id2)
        let tile1 = {...generatedTiles[index1]}
        let tile2 = {...generatedTiles[index2]}
        newTiles[index1] = tile2;
        newTiles[index2] = tile1;
        newTiles.map(tile => tile.selected = false)
        setGeneratedTiles(newTiles)
        setTileSelected({id: null, selected: false })
    }

    const checkSolution = () => {
        const isCorrect = generatedTiles.reduce((curr,tile, i) => {
                if(tile.id !== i){
                    curr = false
                    return curr
                }
                else {
                    curr = true
                    return curr
                }
        }, false)
        setIsSolved(isCorrect)
    }

    return (
        <Col className="puzzlePage__puzzleBox__CurrentImage" >
            
            {!isSolved? generatedTiles.map(tile => {
                return (
                    <Row key={"tileNumber"+tile.id} className={tile.selected? "selectedTile": ""} 
                    onClick={() => handleSelect(tile.id)}>{tile.id}</Row>
                )
            }): ""}
        </Col>
    )
}

export default CurrentImage;