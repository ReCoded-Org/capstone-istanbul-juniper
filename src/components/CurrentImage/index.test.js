import React from "react";
import CurrentImage from "./index.jsx";
import renderer from "react-test-renderer";

describe("CurrentImage Component", () => {
  it("matchess the tree", () => {
    const tree = renderer.create(<CurrentImage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
