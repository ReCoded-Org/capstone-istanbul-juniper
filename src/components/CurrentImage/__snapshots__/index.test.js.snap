// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentImage Component matchess the tree 1`] = `
<div
  className="ant-col puzzlePage__puzzleBox__CurrentImage"
  style={Object {}}
/>
`;
